### PR TITLE
README: Add tutorial on buildbot-worker and change structure

### PR DESCRIPTION
**Pull request by Akira25**
This introduces a tutorial for adidng a buildbot worker to config management.

### CODE REVIEW

File: Tutorials.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+[buildbotworker]
+buildbot-worker-01 ansible_host=85.215.202.123
+buildbot-worker-MYWORKER ansible_host=IP-ADDR
+```
+
+For the `ansible_host` variable, you can give either a (public) IP Address our an hostname like `example.org`.
+
+### Adding variables for buildbot master
+
+asldkfj
```
Comment (Noki) [inline Tutorials.md:48]: gibberish..

Comment (Akira25) [reply Tutorials.md]: right. Adjusted in new commit.

File: Tutorials.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+## Tutorial for provisioning a new machine
+
+Provisioning a new machine happens in two stages:
+
+1. Set up the users and basic installation
+2. Setting it up for its actual purpose
+
+### 1. Set up the basic installation
+
+For this stage, you need to access the machine as root user. Add the machine to the `hosts`-file and write it into the right section to tag its purpose. For defining a machine to serve as a buildbot worker, it should look like something similar to this:
```
Comment (Noki) [inline Tutorials.md:14]: Describe that is required that you have checked out the repository at this point.

give the full path of the hosts file within the repository

File: Tutorials.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+```txt
+[...]
+
+[buildbotworker]
+testmachine ansible_host=example.org
```
Comment (Noki) [inline Tutorials.md:20]: the buildbotworker is a bad example, as it is identical to the example below. choose another one.

File: Tutorials.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+```
+
+After that, you can start the first provision stage with `ansible-playbook play.yml --limit=testmachine --tags=user_provision --user root`. As there are no other users yet, you should not forget the `--user root` option.
+
+### 2. Provision machine for its actual purpose
+
+After that, just run the rest of the playbooks for the given machine, accessing it with your own ssh login: `ansible-playbook play.yml --limit=testmachine`.
```
Comment (Noki) [inline Tutorials.md:28]: This requires a .vaultpass file. This needs to be explained and should be listed as prerequirement.

File: Tutorials.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+For adding your buildbotworker machine to this repo, we need to follow three steps.
+
+### Get your machine known to ansible
+
+To get your machine known to ansible, add it to the `hosts`-file at `inventory/hosts`.
+
+```toml
+[buildbotworker]
+buildbot-worker-01 ansible_host=85.215.202.123
```
Comment (Noki) [inline Tutorials.md:40]: give multiple examples with comments:

```
buildbot-worker-01 ansible_host=85.215.202.123                  # reachable using a public IP
buildbot-worker-02 ansible_host=worker2.example.com             # reachable using a public host
buildbot-worker-akira ansible_host=192.168.16.7                 # reachable only from within martins home using a private IP
buildbot-worker-scherer8 ansible_host=scherer8-buildbot.olsr    # reachable only from within freifunk-net use a jump-host or so...
```

File: Tutorials.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+buildbot-worker-MYWORKER ansible_host=IP-ADDR
+```
+
+For the `ansible_host` variable, you can give either a (public) IP Address our an hostname like `example.org`.
```
Comment (Noki) [inline Tutorials.md:44]: is the public IP/host really required?

File: Tutorials.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+### Adding variables for buildbot master
+
+Please open `inventory/group_vars/all`. There you will find a list with all workers that are allowed to connect to the buildbot master.
```
Comment (Noki) [inline Tutorials.md:48]: This is encrypted. How do you edit it?

File: Tutorials.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
+# Tutorials
+
+In this file, you will find some tutorials describing common workflows for using this repo. We generally assume, that all Hosts run a recent stable Debian version (at time of writing: Debian 11). Our custom roles do mostly rely on this.
+
+## Tutorial for provisioning a new machine
+
+Provisioning a new machine happens in two stages:
+
+1. Set up the users and basic installation
```
Comment (Noki) [inline Tutorials.md:9]: This is already done in a basic debian install, right? It automatically creates a user and a root account. Should you setup an additional user for the worker?

**Review by Noki — CHANGES_REQUESTED** [review]

**Comment by Akira25** [comment]
@Noki I uploaded a new version, which hopefully clarifies things.

File: Tutorials.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
+# Tutorials
+
+In this file, you will find some tutorials describing common workflows for using this repo. We generally assume, that all Hosts run a recent stable Debian version (at time of writing: Debian 11). Our custom roles do mostly rely on this.
```
[the remark this answers is not in the document]
Comment (Akira25) [reply Tutorials.md]: Added in new version.

File: README.md [hunk 10]
```diff
@@ -57,28 +57,3 @@ If you want to help us, you are very welcome.
 The easiest way is to ask `@akira` or `@rtznprmpftl` in our [matrix](https://app.element.io/#/room/#berlin.freifunk.net:matrix.org) chatroom.
 
 Alternatively you can use the GitHub bugtracker, if you discover any issues on the infrastructure side.
-
-## Tutorial for provisioning a new machine
```
[the remark this answers is not in the document]
Comment (Akira25) [reply README.md:61]: Added in new version.

File: Tutorials.md (hunk 2 above)
Comment (Akira25) [reply Tutorials.md]: Added in new version.

File: Tutorials.md (hunk 3 above)
Comment (Akira25) [reply Tutorials.md:35]: Addressed in new version.

File: Tutorials.md (hunk 4 above)
Comment (Akira25) [reply Tutorials.md:43]: It is in the general prerequisites now.

File: Tutorials.md (hunk 5 above)
Comment (Akira25) [reply Tutorials.md]: Added.

File: Tutorials.md (hunk 6 above)
Comment (Akira25) [reply Tutorials.md]: It isn't. added.

File: Tutorials.md (hunk 7 above)
Comment (Akira25) [reply Tutorials.md:65]: The cannonical way of editing this, is to open it with `ansible-vaultpass edit $FILE`. This requires a `.vaultpass`-file though.

File: Tutorials.md (hunk 8 above)
Comment (Akira25) [reply Tutorials.md:24]: There is an additional user for the buildbot worker for improved security reasons, but it gets created by the buildbot-worker role itself. There is no need for further interaction with the machine except for running ansible.

**Comment by Noki** [comment]
The toml formatting looks odd. I would probably change it to a standard code block and than this PR can be merged:
https://github.com/freifunk-berlin/ansible/blob/5010939bfb8634ffd68abd09751b2969ac9d465f/Tutorials.md#1-set-up-the-basic-installation
https://github.com/freifunk-berlin/ansible/blob/5010939bfb8634ffd68abd09751b2969ac9d465f/Tutorials.md#get-your-machine-known-to-ansible

**Review by nicolasberens — APPROVED** [review]

**Comment by Akira25** [comment]
Changed the strange looking markups. Merging due approval.

**Review by Noki — APPROVED** [review]

**Pull request merged**